search-service/src/main/kotlin/com/egm/stellio/search/service/EntityPayloadService.kt

```diff
@@ -424,18 +424,6 @@ class EntityPayloadService(
             // when dealing with an entity update, list of types can be empty if no change of type is requested
             if (currentTypes.sorted() == newTypes.sorted() || newTypes.isEmpty() && allowEmptyListOfTypes)
                 return@either UpdateResult(emptyList(), emptyList())
-            if (!newTypes.containsAll(currentTypes)) {
-                val removedTypes = currentTypes.minus(newTypes)
-                return@either updateResultFromDetailedResult(
-                    listOf(
-                        UpdateAttributeResult(
-                            attributeName = JSONLD_TYPE,
-                            updateOperationResult = UpdateOperationResult.FAILED,
-                            errorMessage = "A type cannot be removed from an entity: $removedTypes have been removed"
-                        )
-                    )
-                )
-            }
 
             val updatedPayload = entityPayload.payload.deserializeExpandedPayload()
```

Comment:
since `payload` has also to be updated, it seems there is no choice but to retrieve the entity from the DB